### ISSUE TITLE
Running dependent flows with parameters

### ISSUE THREAD

**Issue by bolto6** [PrefectHQ/prefect#3098]
Hi!
I dont see in doc, how i can add parameters in `b_flow` from `a_flow` and dir parameter. I think for this i can use local context in `b_flow` from `a_flow`. May be add `upstream_tasks` with parameters? 

Flow A:  
```python
@task
def any_work(a: str) -> str:
    return f'/tmp/{a}'

with Flow('a_flow') as flow:
    name = Parameter('name', default='all')
    result = any_work(name)
```
Flow B:  
```python
@task
def any_work(a: str, b: str) -> str:
    return f'{a}/{b}'

with Flow('b_flow') as flow:
    home = Parameter('home', required=True)
    dir = Parameter('dir', default='any')

    result = any_work(home, dir)
```
Flow C:  
```python
a_flow = FlowRunTask(flow_name='a_flow', wait=True)
b_flow = FlowRunTask(flow_name='b_flow', wait=True)

with Flow('c_flow') as flow:
    name = Parameter('name', default='all')
    dir = Parameter('dir', default='any')

    a_flow_state = a_flow(parameters={'name': name})

    # this error code, but how i can add parameters in `b_flow` from `a_flow` and dir parameter?
    result = b_flow(
        upstream_tasks=[a_flow_state],
        parameters={
            'home': a_flow_state.result,
            'dir': dir,
        },
    )

```

**Comment by bolto6** [comment]
@cicdw maybe this bug?

**Comment by znicholasbrown** [comment]
Hi @bolto6 - responded to your question in Slack as well but posting here for posterity:

My recommendation would be to use an independent [results location](https://docs.prefect.io/core/concepts/results.html#results) to store/retrieve data between your flows or to use the `FlowRunTask` to run the next Flow in your pipeline with the parameters of your choosing. Note that you'll need to call FlowRunTask from within the Flow context, so something like this:

```
from prefect import Flow, task
from prefect.tasks.prefect import FlowRunTask

@task
def any_work(a: str) -> str:
    return f'/tmp/{a}'

@task
def any_work(a: str, b: str) -> str:
    return f'{a}/{b}'


with Flow('a_flow') as flow:
    name = Parameter('name', default='all')
    result = any_work(name)
    FlowRunTask(flow_name="b_flow", parameters={'home': result})

```
etc.

**Comment by dmvieira** [comment]
I was in Slack and people recommend me this issue. I can't retrieve results from a dependent Flow.

Looking in code, I saw that `Client.get_flow_run_info` returns flow without result on Flow `serialized_state`. In other hand, it returns each task state with result as `serialized_state`.

A simple trick to retrieve state when use `StartFlowRun` is append `task_run` result [here](https://github.com/PrefectHQ/prefect/blob/96ef85470872593268c9498b57ac9f0b5a268e01/src/prefect/tasks/prefect/flow_run.py#L180) instead of returning just state or using final task state to Flow final state.

Now `result.state` just returns a string `<Success: "All reference tasks succeeded.">` that say nothing and I can't load result.

Here an example of the issue:
```
from prefect.tasks.prefect import StartFlowRun
from prefect import Flow, task
from prefect.engine.results.local_result import LocalResult
task1 = StartFlowRun(
      flow_name="test_flow",
      project_name="test_project",
      wait=True,
      result=LocalResult(".")
)
with Flow("Call Flow") as flow:
    end_flow = task1()
state = flow.run()
state.result[endflow].result #nothing here, just Success string
```

**Comment by dmvieira** [comment]
I did this workaround to retrieve results:

```
import time
from prefect.tasks.prefect import StartFlowRun
from prefect import Flow, task, Client

task1= StartFlowRun(
      flow_name="test_flow",
      project_name="test_project",
      wait=False
)

with Flow("Call Flow") as flow:
    end_flow = task1()

state = flow.run()
flow_id = state.result[end_flow].result
client = Client()

while not client.get_flow_run_info(flow_id).state.is_finished():
    time.sleep(5)
info = client.get_flow_run_info(flow_id)
last_task = info.task_runs.pop()
res = last_task.state.load_result(last_task.state._result).result

print(res)
```

**Comment by peterroelants** [comment]
I'm using the following Task to get the result of a `StartFlowRun`:

```python
class GetFlowResult(prefect.Task):
    """
    Get flow result.
    The flow's result depends on a specific task with a given slug.
    """
    def __init__(self, target_task_slug: str, **kwargs):
        self.target_task_slug = target_task_slug
        super().__init__(**kwargs)

    def run(self, flow_run_state: state.State) -> Any:
        flow_run_id = str(flow_run_state).split(" ")[0]
        client = prefect.Client()
        flow_run_info = client.get_flow_run_info(flow_run_id)
        target_task_run = get_target_task_run_from_slug(target_slug=self.target_task_slug, task_runs=flow_run_info.task_runs)
        res = target_task_run.state.load_result(target_task_run.state._result).result
        return res


def get_target_task_run_from_slug(target_slug: str, task_runs: List[TaskRunInfoResult]) -> TaskRunInfoResult:
    """Get a given task based on the task's slug from a list of task runs."""
    target_task_runs = list(filter(lambda t: t.task_slug and target_slug in t.task_slug, task_runs))
    return target_task_runs[0]
```

See also [this Slack thread](https://prefect-community.slack.com/archives/CL09KU1K7/p1622645858327700).

**Comment by madkinsz** [comment]
Thanks for sharing your task @peterroelants -- this should be resolved by #4563

**Comment by peterroelants** [comment]
@madkinsz If I understand correctly #4563 is not yet in the 0.14.21 release, but might be in the next one?

**Comment by madkinsz** [comment]
Yeah it'll be released after #4434 is merged which I have slated for the next release
